qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -509,63 +533,404 @@ def _build_eom_matrices(
+
+            # 1. Build excitation operators O_l such that O_l |0> = |l>
+            excitations_ops_reduced = self._build_excitation_operators(
+                expansion_basis_data, reference_state, expansion_coefs_rescaled
+            )
+
+            # 2. Prepare observables O_k^\dag @ Aux @ O_l
+            op_aux_op_dict = self._prepare_excited_states_observables(
+                untap_aux_ops, excitations_ops_reduced, size
+            )
+
+            # 3. Measure observables
+            tap_op_aux_op_dict = self.qubit_converter.symmetry_reduce_clifford(op_aux_op_dict, True)
+            aux_measurements = estimate_observables(
+                self._estimator, reference_state[0], tap_op_aux_op_dict, reference_state[1]
+            )
+
+            # 4. Format aux_operators_eigenvalues
+            indices_diag = np.diag_indices(size + 1)
+            indices_diag_as_list = list(zip(indices_diag[0], indices_diag[1]))
+            for indice in indices_diag_as_list:
+                aux_operators_eigenvalues[indice] = {}
+                for aux_name in untap_aux_ops.keys():
+                    aux_operators_eigenvalues[indice][aux_name] = aux_measurements.get(
+                        (aux_name, indice[0], indice[1]), (0.0, {})
+                    )
+
+            # 5. Format transition_amplitudes
+            indices_offdiag = np.triu_indices(size + 1, k=1)
+            indices_offdiag_as_list = list(zip(indices_offdiag[0], indices_offdiag[1]))
+            for indice in indices_offdiag_as_list:
+                transition_amplitudes[indice] = {}
+                for aux_name in untap_aux_ops.keys():
+                    transition_amplitudes[indice][aux_name] = aux_measurements.get(
+                        (aux_name, indice[0], indice[1]), (0.0, {})
+                    )
+
+        return aux_operators_eigenvalues, transition_amplitudes
+
+    def _build_qeom_result(
+        self,
+        problem,
+        groundstate_result,
+        expansion_coefs,
+        energy_gaps,
+        h_mat,
+        s_mat,
+        h_mat_std,
+        s_mat_std,
+        aux_operators_eigenvalues,
+        transition_amplitudes,
+        gammas_square,
+    ) -> ElectronicStructureResult:
+
+        qeom_result = QEOMResult()
+        qeom_result.ground_state_raw_result = groundstate_result.raw_result
+        qeom_result.expansion_coefficients = expansion_coefs
+        qeom_result.excitation_energies = energy_gaps
+
+        qeom_result.m_matrix = h_mat[: len(h_mat) // 2, : len(h_mat) // 2]
+        qeom_result.v_matrix = s_mat[: len(h_mat) // 2, : len(h_mat) // 2]
+        qeom_result.q_matrix = h_mat[len(h_mat) // 2 :, : len(h_mat) // 2]
+        qeom_result.w_matrix = s_mat[len(h_mat) // 2 :, : len(h_mat) // 2]
+        qeom_result.m_matrix_std = h_mat_std[0, 0]
+        qeom_result.v_matrix_std = s_mat_std[0, 0]
+        qeom_result.q_matrix_std = h_mat_std[0, 1]
+        qeom_result.w_matrix_std = s_mat_std[0, 1]
```

Comment:
This is good, thank you :+1: 

@manoelmarques what would the deprecation strategy look like here, if I want to make these read-only properties of the result object. Doing this would allow us to put this computation logic into the property getters and ensure that the matrices stored in the result are always consistent (plus it simplifies the code here a bit).

Reply:
If you put the calculating logic in the init method for each property, then you just need the deprecation msg in the setters. This way the user still can use the deprecated setter which would override the calculation done in the init.

Reply:
Thanks, that's a good idea :+1: @Anthony-Gandon could you do that, please?

Reply:
As discussed with @mrossinek , I could not put the calculating logic in the init method. I changed the getters instead.